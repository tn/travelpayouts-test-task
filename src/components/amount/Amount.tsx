import * as React from 'react'

interface IAmountProps {
  currency: string
  amount: number
}

const Amount: React.SFC<IAmountProps> = ({ currency, amount }) => {
  const res = amount.toLocaleString(
    'ru-RU',
    {
      style: 'currency',
      currency: currency.toUpperCase()
    })

  return <React.Fragment>{res}</React.Fragment>
}

export default Amount
