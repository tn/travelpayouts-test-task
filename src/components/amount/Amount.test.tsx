import * as React from 'react'
import * as Adapter from 'enzyme-adapter-react-16'
import { configure, shallow } from 'enzyme'
import * as renderer from 'react-test-renderer'
import Amount from './Amount'

configure({ adapter: new Adapter() })

describe('Amount', () => {
  const props = {
    currency: 'rub',
    amount: 100000
  }

  it('should be rendered', () => {
    const wrapper = renderer.create(<Amount {...props} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should format amount', () => {
    const wrapper = shallow(<Amount {...props} />)

    expect(wrapper.text()).toEqual('RUBÂ 100,000.00')
  })
})
