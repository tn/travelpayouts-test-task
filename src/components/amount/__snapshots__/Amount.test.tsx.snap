// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amount should be rendered 1`] = `"RUBÂ 100,000.00"`;
